framework/src/userobjects/SidesetAroundSubdomainUpdater.C

```diff
@@ -0,0 +1,184 @@
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SidesetAroundSubdomainUpdater.h"
+#include "DisplacedProblem.h"
+#include "Assembly.h"
+
+#include "libmesh/parallel_algebra.h"
+#include "libmesh/parallel_sync.h"
+
+registerMooseObject("MooseApp", SidesetAroundSubdomainUpdater);
+
+InputParameters
+SidesetAroundSubdomainUpdater::validParams()
+{
+  InputParameters params = DomainUserObject::validParams();
+  params.addClassDescription("Sets up a boundary between inner_subdomains and outer_subdomains");
+  params.addParam<std::vector<SubdomainName>>("inner_subdomains",
+                                              "Subdomains that own the boundary");
+  params.addParam<std::vector<SubdomainName>>("outer_subdomains",
+                                              "Subdomains on the outside of the boundary");
+  params.addParam<bool>("assign_surface_sides",
+                        true,
+                        "Assign sides of elements im `inner_subdomains` that have no neighbor.");
+  params.addRequiredParam<BoundaryName>("update_sideset_name",
+                                        "The name of the sideset to be updated.");
+  return params;
+}
+
+SidesetAroundSubdomainUpdater::SidesetAroundSubdomainUpdater(const InputParameters & parameters)
+  : DomainUserObject(parameters),
+    _comm(_mesh.comm()),
+    _pid(_comm.rank()),
+    _displaced_problem(_fe_problem.getDisplacedProblem().get()),
+    _neighbor_side(_assembly.neighborSide()),
+    _assign_surface_sides(getParam<bool>("assign_surface_sides")),
+    _boundary_name(getParam<BoundaryName>("update_sideset_name")),
+    _boundary_id(_mesh.getBoundaryID(_boundary_name)),
+    _boundary_info(_mesh.getMesh().get_boundary_info()),
+    _displaced_boundary_info(
+        _displaced_problem ? &_displaced_problem->mesh().getMesh().get_boundary_info() : nullptr)
+{
+  // subdomains
+  const auto inner_subdomains = getParam<std::vector<SubdomainName>>("inner_subdomains");
+  const auto outer_subdomains = getParam<std::vector<SubdomainName>>("outer_subdomains");
+  for (const auto & name : inner_subdomains)
+    if (!MooseMeshUtils::hasSubdomainName(_mesh.getMesh(), name))
+      paramError("inner_subdomains", "The block '", name, "' was not found in the mesh");
+  for (const auto & name : outer_subdomains)
+    if (!MooseMeshUtils::hasSubdomainName(_mesh.getMesh(), name))
+      paramError("outer_subdomains", "The block '", name, "' was not found in the mesh");
+
+  for (const auto id : MooseMeshUtils::getSubdomainIDs(_mesh.getMesh(), inner_subdomains))
+    _inner_ids.insert(id);
+  for (const auto id : MooseMeshUtils::getSubdomainIDs(_mesh.getMesh(), outer_subdomains))
+    _outer_ids.insert(id);
+
+  // save boundary name
+  _boundary_info.sideset_name(_boundary_id) = _boundary_name;
+  _boundary_info.nodeset_name(_boundary_id) = _boundary_name;
```

Comment:
You might say in the documentation string that if the boundary information doesn't yet exist, it will be created

Reply:
Added to the parameter doc